logs_sdk/lib/opentelemetry/sdk/logs/log_record_processor.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # LogRecordProcessor describes a duck type and provides synchronous no-op hooks for when a
+      # {LogRecord} is started or when a {LogRecord} is ended. It is not required to subclass this
+      # class to provide an implementation of LogRecordProcessor, provided the interface is
+      # satisfied.
+      class LogRecordProcessor
+        # Called when a {LogRecord} is emitted. Subsequent calls are not
+        # permitted after shutdown is called.
+        # @param [LogRecord] log_record The emitted {LogRecord}
+        # @param [Context] context The resolved Context
```

Comment:
What is the "resolved Context" here? Who does the resolution? (Note that I haven't read this part of the spec or all of this PR, or the logs API, so maybe this is answered clearly somewhere else - just asking as a naive reviewer.)

Reply:
The "resolved Context" here is coming from the Logs SDK spec description for the method arguments. 

https://opentelemetry.io/docs/specs/otel/logs/sdk/#onemit

The description has a bit more detail to include that might clear things up. I've updated the description to include the rest of the text in: 5528277401c8354cfab8ade8a17f4d65b20f7dd9

Reply:
I'm not sure who does the resolution. I'm also not quite sure how we'll use this argument. 

I'll consult some of the implementations in other languages and come back with more info.

Reply:
@fbogsany  - I can't find any examples where the `context` argument is used in other languages. 
* Python doesn't include it anywhere. 
* PHP and Java have it as an argument but do not use it in the `onEmit` functions. 
* Javascript doesn't include it as an argument in their Simple or Batch processors, but the duck type includes it as an argument.

So I don't have a great answer about who does the resolution. I didn't end up needing this argument in the Simple or Batch processors because I used the SpanContext to pull `trace_id`, `span_id`, and `trace_flags`. Those are the only context-related attributes in the [Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/#log-and-event-record-definition).

Do you have a sense of what approach you'd prefer Ruby take?